### DISCUSSION TITLE
Observations from the Ground

### DISCUSSION THREAD

**Discussion by karsood** [Sunbird-cQube/community#41] [General]
Hello team,

I work on the program team at EkStep and have been closely involved in implementing the Saral and cQube building blocks across states. I have had multiple discussions with tech partners, govt. stakeholders and leaders towards successful adoption.
Through this post, I am sharing some of the observations I have had to further enhance cQube and Saral. The document attached has different usecases listed with details for them.

I would like to further engage with the larger team to take this discussion forward and see what possibilities can be unlocked through Saral and cQube.

[LINK](https://drive.google.com/file/d/1yxlg0N62WMEfUMJpinLB_h8pfn43_S3s/view?usp=sharing)
Regards,
Karan Sood
[cQube+Saral_MRD.pdf](https://github.com/Sunbird-cQube/community/files/8649260/cQube%2BSaral_MRD.pdf)

**Comment by SreenivasNimmagadda** [comment]
Hi Karan,

Thanks for sharing your observations.  
As you know, cQube is a lightweight tool to generate pre-defined metrics and creates visualizations of these metrics on a regular basis. 

Please find the below possibilities for the points which you mentioned in the PDF. 

1. Students filled OMR sheets to be scanned by Teachers at the School: 
Scanning OMR sheets into cQube cannot be done directly. The data needs to be provided and ingested as per cQube standard formats.

2. MCQ options:
Scanning OMR sheets into cQube cannot be done directly. The  periodic assessment test data which contains obtained marks needs to be provided and ingested as per cQube standard formats.

3. Non-academic Data Scanning: 
Scanning OMR sheets into cQube cannot be done directly.

4. Student Report card: 
cQube does not support the student level reporting. So, This is out of cQube scope.

5. Courses Report card: 
As above mentioned cQube is not designed Student or teacher level, This item is also out of cQube scope.

6. Role-based access to cQube: 
The current cQube is having the capacity of User level based accessing of the cQube reports. The current Access levels we have in cQube are 

State level users, 
District level users, 
Block level users, 
cluster level users,
school level users

For example, If the user is belongs to a district then the user can be able to access only their district information as well as the lower levels of their districts.
If the user is belongs to a Block level then the user may able to see their block level information as well as the lower levels of their block.

7. Ability to export report images for presentations: 
current cQube is not having this functionality.  We will look into the possibilities to implement this functionality. If possible we will be able to consider with appropriate development timelines.

8. Additional report generation capability within cQube similar to HawkEye for DIKSHA: 
Configurability of the Data sources and Visualization is the solution for this point. This functionality is still in the architecting stage. Will come up with the functionality completion time lines once the architecture is finalized.

Hope I given the clarity to your points.

Thanks.

**Comment by dileep-gadiraju** [comment]
Hi @karsood ,

Thanks for sharing observations from the ground.
Please find the response below w.r.t. Saral.

1. Students filled out OMR sheets to be scanned by Teachers at the School.  The exam formats are of various types ranging in a number of questions from 5 to 100+ questions. We need to cater to OMR designs that can handle this variation in a single sheet:   **Response: currently, on one of the layouts, we are covering 90 questions for 3 students. So if it's single student 90 single omr questions can be accommodated on high-end mobiles - Android 10 and above. As the scanning is totally based on the compute and scanning power of mobile, layouts should be designed accordingly.**
2. MCQ options: As students fill out responses in OMRs, they must be given the choice to choose one of four options (A, B, C, D). Saral should be able to collect the information on the basis of the option being darkened and cQube should be able to evaluate the option chosen and provide either 0 marks (wrong option) or ‘x’ marks (right option). UP and AS have expressed this requirement.  **Response: Saral supports collecting MCQ but it sends an index of the option instead of values as of today.** 
3.  Non-academic Data Scanning Assessments are only one record that teachers have to maintain. Records like attendance, student performance cards, MDM, article distribution, etc. prove to be time-consuming activities. Saral has the potential to aid in data collection for all these records. VKA in Gorakhpur is looking forward to implementing Saral in non-academic aspects. Assam has School Assessment formats that are filled to adjudicate school performance under Gunotsav.  **Response: yes currently  Handwritten digits and omr based layouts can be supported for non-academic use cases provided minimum layout specifications of dedicated boxes for handwritten digits, omr circle, corner detection dark circles are followed.**

**Reply by SreenivasNimmagadda** [reply]
Hi @dileep-gadiraju, Thanks for giving the additional information regarding the Saral responsibilities. 

@karsood the above points mentioned by @dileep-gadiraju are the Saral responsibilities. As Saral also a part of Sunbird-ED these points were considered. 

Thanks.
